mri/operators/fourier/non_cartesian.py

```diff
@@ -378,9 +385,127 @@ def adj_op(self, x):
+            receiver coils acquisition. If n_coils > 1, please organize data as
+            n_coils X data_per_coil
+    """
+    def __init__(self, samples, shape, n_coils=1, density_comp=None,
+                 kernel_width=3, sector_width=8, osf=2, balance_workload=True,
+                 smaps=None):
+        """ Initilize the 'NUFFT' class.
+
+        Parameters
+        ----------
+        samples: np.ndarray
+            the mask samples in the Fourier domain.
+        shape: tuple of int
+            shape of the image
+        n_coils: int
+            Number of coils used to acquire the signal in case of multiarray
+            receiver coils acquisition
+        density_comp: np.ndarray default None.
+            k-space weighting, density compensation, if not specified
+            equal weightage is given.
+        kernel_width: int default 3
+            interpolation kernel width (usually 3 to 7)
+        sector_width: int default 8
+            sector width to use
+        osf: int default 2
+            oversampling factor (usually between 1 and 2)
+        balance_workload: bool default True
+            whether the workloads need to be balanced
+        smaps: np.ndarray default None
+            Holds the sensitivity maps for SENSE reconstruction
+        """
+        if (n_coils < 1) or (type(n_coils) is not int):
+            raise ValueError('The number of coils should be an integer >= 1')
+        self.n_coils = n_coils
+        self.shape = shape
+        if samples.min() < -0.5 or samples.max() >= 0.5:
+            warnings.warn("Samples will be normalized between [-0.5; 0.5[")
+            samples = normalize_frequency_locations(samples)
+            self.samples = samples
+        if density_comp is None:
+            density_comp = np.ones(samples.shape[0])
+        if smaps is not None:
+            self.uses_smaps = False
+        else:
+            self.uses_smaps = True
+        self.operator = NUFFTOp(
+            np.reshape(samples, samples.shape[::-1], order='F'),
+            shape,
+            n_coils,
+            smaps,
+            density_comp,
+            kernel_width,
+            sector_width,
+            osf,
+            balance_workload
+        )
+
+    def op(self, image):
+        """ This method calculates the masked non-cartesian Fourier transform
+        of a 2D / 3D image.
+
+        Parameters
+        ----------
+        image: np.ndarray
+            input array with the same shape as shape.
+
+        Returns
+        -------
+        np.ndarray
+            Non Uniform Fourier transform of the input image.
+        """
+        if self.n_coils > 1:
+            coeff = self.operator.op(np.asarray(
+                [np.reshape(image_ch.T, image_ch.size) for image_ch in image]
+            ).T)
+        else:
+            coeff = self.operator.op(np.reshape(image.T, image.size))
+            coeff = coeff[0]
+        return np.squeeze(coeff)
```

Comment:
why do you need to squeeze at the end of both ops?

Reply:
The C++ is generalized for 2D and 3D and thus always returns Num_channel x Nx x Ny x Nz. Thus for 2D case it is very important to squeeze the data up as Nz would be 1. (Same for multichannel)

Reply:
ok makes sense. I guess you don't need the `coeff[0]` just above then.

Reply:
Well for multichannel, I know it must be coeff[0] as Num_channels is 1. 
I just want to try to get away from squeezing as much as possible so that code is more readable

Reply:
sure ok. I say let's comment on all this in the code.

Reply:
I just realized that we dont need squeeze for this case, as we used coeff[0]. Sorry that I missed that. I have documented code in other places where I used it and also here for coeff[0]